src/WebJobs.Script.WebHost/Diagnostics/DictionaryLoggerScope.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    /// <summary>
+    /// This class is identical to the internal class in WebJobs.Host. It is copied here so we can
```

Comment:
Instead of copying this, why not make it public?

Alternatively, we could make the argument that script is a friend assembly of core and add a: `[assembly: InternalsVisibleTo("Microsoft.Azure.WebJobs.Script")]` in the core sdk.

Reply:
It's really great to see the community weighing in on our PRs :-)

Yeah, I want to make this better. Exposing it publicly (or even as an friend) as-is won't work like we want b/c it tracks the scope internally as a static field. That means if two classes use this class, you'll get a bunch of calls to modify the same `AsyncLocal`, which isn't what we want. I'd like to refactor it so the logic can be re-used -- I think we're using it in 3 places now. I've opened https://github.com/Azure/azure-webjobs-sdk/issues/1451 so I can follow-up in a subsequent change.

Reply:
Yeah I don't like the copy either - there's gotta be a better way